contributors/devel/conformance-tests.md

```diff
@@ -0,0 +1,155 @@
+# Conformance Testing in Kubernetes
+
+The Kubernetes conformance test suite is a set of testcases, currently a
+subset of the integration/e2e tests, that the Architecture SIG has approved
+to define the core set of interoperable features that all Kubernetes
+deployments must support.
+
+Contributors must write and submit e2e tests first (approved by owning Sigs). 
+Once the new tests prove to be stable in CI runs, later create a follow up PR 
+to add the test to conformance. This approach also decouples the development 
+of useful tests from their promotion to conformance.
+
+A conformance test verifies the expected functionality works as a user might encounter it in the wild, 
+and tests should begin by covering the most important and visible aspects of the function.
+
+### Conformance Test Requirements
+
+A test is eligible for promotion to conformance if it meets the following requirements:
+
+- testing GA feature (not alpha or beta APIs, nor deprecated features)
+- must be portable (not dependent on provider-specific capabilities or on the public internet)
+- cannot test a feature which obviously cannot be supported on a broad range of platforms 
+(i.e. testing of multiple disk mounts, GPUs, high density)
+- cannot test an optional feature (e.g. not policy enforcement)
+- should be non-privileged (neither root on nodes, network, nor cluster)
+- cannot rely on any particular non-standard file system permissions granted to containers or users 
+(i.e. sharing writable host /tmp with a container)
+- should be stable and run consistently
+- cannot skip providers (there should be no Skip like directives added to the test), 
+especially in the Nucleus or Application layers as described 
+[here](https://github.com/kubernetes/community/blob/master/contributors/devel/architectural-roadmap.md).
+- cannot test cloud provider specific features (i.e. GCE monitoring, S3 Bucketing, ...)
+- should work with default settings for all configuration parameters 
+(example: the default list of admission plugins should not have to be tweaked for passing conformance).
+- cannot rely on any binaries that are not required for the
+linux kernel or for a kubelet to run (i.e. git)
+
+### Conformance Test Version Skew Policy
+
+As each new release of Kubernetes provides new functionality, the subset of
+tests necessary to demonstrate conformance grows with each release. Conformance
+is thus considered versioned, with the same backwards compatibility guarantees
+as laid out in [our versioning policy](/contributors/design-proposals/release/versioning.md#supported-releases-and-component-skew).
+Conformance tests for a given version should be run off of the release branch
+that corresponds to that version. Thus `v1.2` conformance tests would be run
+from the head of the `release-1.2` branch. eg:
+
+ - A v1.3 development cluster should pass v1.1, v1.2 conformance tests
```

Comment:
What does "1.3 development cluster" mean? A cluster built from the master branch after the 1.2 release was cut but before the 1.3 release was cut?

Note that kubectl officially only supports one release of skew.

Reply:
yes

where are the docs on kubectl skew? I wasn't aware it was that limited

Reply:
Not the most obvious location, I know, but the origin of this doc predates kubernetes.io:
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/release/versioning.md#supported-releases-and-component-skew

Reply:
Yes.